exhibitions/src/main/kotlin/edu/artic/exhibitions/ExhibitionDetailViewModel.kt

```diff
@@ -0,0 +1,92 @@
+package edu.artic.exhibitions
+
+import com.fuzz.rx.bindTo
+import com.fuzz.rx.disposedBy
+import edu.artic.base.utils.DateTimeHelper
+import edu.artic.db.daos.ArticDataObjectDao
+import edu.artic.db.models.ArticExhibition
+import edu.artic.viewmodel.NavViewViewModel
+import edu.artic.viewmodel.Navigate
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import javax.inject.Inject
+
+class ExhibitionDetailViewModel
+@Inject
+constructor(dataObjectDao: ArticDataObjectDao)
+    : NavViewViewModel<ExhibitionDetailViewModel.NavigationEndpoint>() {
+    sealed class NavigationEndpoint {
+        class ShowOnMap(val exhibition: ArticExhibition) : NavigationEndpoint()
+        class BuyTickets(val url: String) : NavigationEndpoint()
+    }
+
+    val imageUrl: Subject<String> = BehaviorSubject.create()
+    val title: Subject<String> = BehaviorSubject.createDefault("test")
+    val metaData: Subject<String> = BehaviorSubject.createDefault("")
+    val showOnMapButtonText: Subject<String> = BehaviorSubject.createDefault("Show on Map") // TODO: replace when special localizer is done
+    val buyTicketsButtonText: Subject<String> = BehaviorSubject.createDefault("Buy Tickets")// TODO: replace when special localizer is done
+    val description: Subject<String> = BehaviorSubject.createDefault("")
+    val throughDate: Subject<String> = BehaviorSubject.createDefault("")
+
+    var ticketsUrl: String? = null
+    var exhibition: ArticExhibition? = null
+
+    init {
+        dataObjectDao.getDataObject()
+                .filter { it.ticketsUrl != null }
+                .subscribe {
+                    ticketsUrl = it.ticketsUrl!!
```

Comment:
explicit unwrapping means something is wrong. use ` ?: ""` or something

Reply:
fixed on another branch.

Reply:
also there is no posibility that it is a null as we filter it. just rx being rx.